packages/manager/core/application/src/ovh-tracking.tsx

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useLayoutEffect, useState, useRef } from 'react';
+import { useLocation } from 'react-router-dom';
+import { AT_INTERNET_LEVEL2 } from '@ovh-ux/ovh-at-internet';
+import { useShell } from '.';
+
+// Send hit with data-tracking attributes
+const ODS_COMPONENTS = ['OSDS-BUTTON', 'OSDS-LINK'];
+// Send hit when value attributes change
+const ODS_COMPONENTS_GROUP = [
+  'osds-select',
+  'osds-radio-group',
+  'osds-checkbox',
+];
+
+export default function OvhTracking() {
+  const location = useLocation();
+  const shell = useShell();
+  const { tracking, environment } = shell;
+  const env = environment.getEnvironment();
+  const [locationPath, setLocationPath] = useState(location);
+  const myStateRef = useRef(locationPath);
+
+  const trackLevel2 = (universe: string) => {
+    const result = Object.keys(AT_INTERNET_LEVEL2).filter((element) => {
+      if (
+        AT_INTERNET_LEVEL2[element]
+          .toLowerCase()
+          .indexOf(universe.toLowerCase()) > -1
+      ) {
+        return element;
+      }
+      return false;
+    });
+    return result ? result[0] : '0';
+  };
+
+  // Use trackingPage to send action load
+  const OvhTrackPage = (loc: any) => {
+    const path = loc?.pathname.split('/')[1];
+    const partsReplace = loc.pathname.substring(1).replaceAll('/', '::');
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${universe}::${applicationName}::${partsReplace ||
+        'homepage'}`;
+      tracking.trackPage({
+        name,
+        level2: trackLevel2(universe),
+        page_category: path || 'homepage',
+      });
+    });
+  };
+
+  useEffect(() => {
+    tracking.init(false);
+  }, []);
+
+  useEffect(() => {
+    // When location change, send tracking load page
+    OvhTrackPage(location);
+    myStateRef.current = location;
+    setLocationPath(location);
+  }, [location]);
+
+  const ovhTrackingSendCLick = (value: string) => {
+    const path = myStateRef.current.pathname.split('/')[1];
+    env.then((response) => {
+      const { applicationName, universe } = response;
+      const name = `${applicationName}::${path || 'homepage'}::${value}`;
+      const page = `${universe}::${applicationName}::${path || 'homepage'}`;
+      tracking.trackClick({
+        name,
+        page: { name: page },
+        complete_page_name: page,
```

Comment:
**question:** Are you sure `trackClick` expects `page` and `complete_page` keys?
I dont see such keys on https://www.npmjs.com/package/ng-at-internet/v/3.1.1

Reply:
Its a new request from tracking team